File: usb_android/lib/src/my_usb_interface.dart
import 'package:usb_platform_interface/usb_platform_interface.dart';

import 'my_api.dart';
import 'my_object.dart';

final myUsbInterfaceApi = MyUsbInterfaceHostApi();

class MyUsbInterface extends MyObject implements UsbInterface {
  MyUsbInterface(super.hashCode);

  @override
  Future<int> getAlternateSetting() {
    return myUsbInterfaceApi.getAlternateSetting(hashCode);
  }

  @override
  Future<int> getEndpointCount() {
    return myUsbInterfaceApi.getEndpointCount(hashCode);
  }

  @override
  Future<int> getId() {
    return myUsbInterfaceApi.getId(hashCode);
  }

  @override
  Future<int> getInterfaceClass() {
    return myUsbInterfaceApi.getInterfaceClass(hashCode);
  }

  @override
  Future<int> getInterfaceProtocol() {
    return myUsbInterfaceApi.getInterfaceProtocol(hashCode);
  }

  @override
  Future<int> getInterfaceSubclass() {
    return myUsbInterfaceApi.getInterfaceSubclass(hashCode);
  }

  @override
  Future<String?> getName() {
    return myUsbInterfaceApi.getName(hashCode);
  }
}

File: usb_android/android/src/main/kotlin/dev/yanshouwang/usb_android/MyUsbInterface.kt
package dev.yanshouwang.usb_android

import android.hardware.usb.UsbInterface

class MyUsbInterface(private val collector: MyCollector) : MyUsbInterfaceHostApi {
    override fun getId(hashCode: Long): Long {
        val usbInterface = collector.instanceOf(hashCode) as UsbInterface
        return usbInterface.id.toLong()
    }

    override fun getAlternateSetting(hashCode: Long): Long {
        val usbInterface = collector.instanceOf(hashCode) as UsbInterface
        return usbInterface.alternateSetting.toLong()
    }

    override fun getInterfaceClass(hashCode: Long): Long {
        val usbInterface = collector.instanceOf(hashCode) as UsbInterface
        return usbInterface.interfaceClass.toLong()
    }

    override fun getInterfaceSubclass(hashCode: Long): Long {
        val usbInterface = collector.instanceOf(hashCode) as UsbInterface
        return usbInterface.interfaceSubclass.toLong()
    }

    override fun getInterfaceProtocol(hashCode: Long): Long {
        val usbInterface = collector.instanceOf(hashCode) as UsbInterface
        return usbInterface.interfaceProtocol.toLong()
    }

    override fun getName(hashCode: Long): String? {
        val usbInterface = collector.instanceOf(hashCode) as UsbInterface
        return usbInterface.name
    }

    override fun getEndpointCount(hashCode: Long): Long {
        val usbInterface = collector.instanceOf(hashCode) as UsbInterface
        return usbInterface.endpointCount.toLong()
    }
}